modify the column type change doc (#5787)
pingcap/docs: sql-statements/sql-statement-change-column.md

The `ALTER TABLE.. CHANGE COLUMN` statement changes a column on an existing table. The change can include both renaming the column, and changing the data type to a compatible type.

> Since v5.1.0, TiDB has supported changes of data types for Reorg data, including but not limited to:

Reviewer: ```suggestion
Since v5.1.0, TiDB has supported changing the Reorg data type, including but not limited to:
```

> - Change from varchar to bigint
> - Modify decimal precision
> - Length compression from varchar(10) to varchar(5)

Reviewer: ```suggestion
- Changing `VARCHAR` to `BIGINT`
- Modifying the `DECIMAL` precision
- Compressing the length of `VARCHAR(10)` to `VARCHAR(5)`
```

```

```
ERROR 1105 (HY000): can't run multi schema change
```

{{< copyable "sql" >}}

```sql
CREATE TABLE t (a int primary key);
ALTER TABLE t CHANGE COLUMN a a VARCHAR(10);
```

```
ERROR 8200 (HY000): Unsupported modify column: column has primary key flag
```

{{< copyable "sql" >}}

```sql
CREATE TABLE t (c1 INT, c2 INT, c3 INT) partition by range columns(c1) ( partition p0 values less than (10), partition p1 values less than (maxvalue));
ALTER TABLE t CHANGE COLUMN c1 c1 DATETIME;
```

```
ERROR 8200 (HY000): Unsupported modify column: table is partition table
```

{{< copyable "sql" >}}

```sql
CREATE TABLE t (a INT, b INT as (a+1));
ALTER TABLE t CHANGE COLUMN b b VARCHAR(10);
```

```
ERROR 8200 (HY000): Unsupported modify column: column is generated
```

{{< copyable "sql" >}}

```sql
CREATE TABLE t (a DECIMAL(13, 7));
ALTER TABLE t CHANGE COLUMN a a DATETIME;
```

```
ERROR 8200 (HY000): Unsupported modify column: change from original type decimal(13,7) to datetime is currently unsupported yet
```

## MySQL compatibility

* Making multiple changes in a single `ALTER TABLE` statement is not currently supported.

> * Changes of data type on primary key columns are not supported.

Reviewer: ```suggestion
* Changes of data types on primary key columns are not supported.
```

* Changes of column types on partitioned tables are not supported.
* Changes of column types on the generated columns are not supported.

> * Changes of some data types (for example, some TIME types, Bit, Set, Enum, JSON, etc.) are not supported due to some compatibility issues of TIDB cast function with MySQL's behavior.

Reviewer: ```suggestion
* Changes of some data types (for example, some TIME types, Bit, Set, Enum, JSON) are not supported due to some compatibility issues of the `cast` function's behavior between TiDB and MySQL.
```
